### ISSUE TITLE
stack sdist some-sub-dir fails with master

### ISSUE THREAD

**Issue by snoyberg** [commercialhaskell/stack#3919]
This occurs on commit 3ac63288ca92c20dfb62274d1f3286fad6d686bf. I have the `http-client` repo checked out locally. Example output:

```
$ stack --version
Version 1.7.0, Git revision 6b451df44a0d17b0933d37d3d0cea4335d61a532 (dirty) (5713 commits) x86_64 hpack-0.21.2
$ stack sdist http-client
Getting file list for /Users/michael/Documents/http-client/http-client/
Building sdist tarball for /Users/michael/Documents/http-client/http-client/
Checking package 'http-client' for common mistakes
./http-client.cabal: openBinaryFile: does not exist (No such file or directory)
```

By contrast:

```
$ stack-1.6.5 --version
Version 1.6.5, Git revision 24ab0d6ff07f28276e082c3ce74dfdeb1a2ca9e9 (5514 commits) x86_64 hpack-0.20.0
$ stack-1.6.5 sdist http-client
Getting file list for /Users/michael/Documents/http-client/http-client/
Building sdist tarball for /Users/michael/Documents/http-client/http-client/
Checking package 'http-client' for common mistakes
Wrote sdist tarball to /Users/michael/Documents/http-client/http-client/.stack-work/dist/x86_64-osx/Cabal-2.0.1.0/http-client-0.5.11.tar.gz
```

I suspect that the upgrade to Cabal 2.2 may have caused this. Investigating.

**Comment by snoyberg** [comment]
Upstream issue opened: https://github.com/haskell/cabal/issues/5206

**Comment by 23Skidoo** [comment]
Can confirm that the fix for https://github.com/haskell/cabal/issues/5206 is included in the `Cabal-2.2.0.1` release.
